ported editlog to lib2 and added condition state editing

following5 opened the pull request:
This is **beta** code, for the case someone would like to help testing. Deployment to test.opencaching.de would be great. Don't deploy to procuction site yet!

* complete overhaul of editlog.php
* minor adjustments to log.php

Will produce merge conficts with #538, so one of them will eventually need a rebase.

Branch is based on PR #551 to avoid even more merge conflicts.

_(Ich mache dann noch Redmine #987 und dann ist meine Todo-Liste erst mal leer.  ;)_

following5 committed:
minimal port of editlog to lib2; updates #790

following5 committed:
editlog formal refactoring, no code changes

following5 committed:
full smarty port of editlog; updates #790

following5 committed:
merged log_cache and editlog templates and added cache condition
state editing to editlog; updates #1050

teiling88 commented:
Ich weiß nicht ob sich der Aufwand hier lohnt. Wir setzen gerade Seite für Seite auf die neue Struktur mit Symfony etc um. 
Was meinst Du?

following5 commented:
Für die Support- und Datenpflegearbeit ist dieses Feature wichtig, d.h. vor allem für mic@, dogesu und mich. Nicht umsonst hat mic@ im Laufe eines Jahres gleich drei Redmine-Ticketes dazu aufgemacht. ;-) Wir loggen häufig Hinweise zum Cache- und Listingstatus, und es kommt öfter mal vor dass man das Statusflag vergisst. Dazu muss man derzeit ein zweites Log einstellen, was erneute Benachrichtiungsmails produziert, mit denen man die Beobachter und vor allem den Owner nervt. (Was auch unserem Grundsatz zuwider läuft, die Support-Emails an die Owner auf das notwendige Minimum zu begrenzen.)

Die anderen Nutzer werden es nicht so häufig brauchen.

Mit diesem Change würden auch weitere Inkonsistenzen zwischen Log-einstellen und Log-bearbeiten behoben, z.B. bessere Erläuterung zum Datumsfeld auch bei editlog und Ausblenden eines unbenötigten Passwort-Eingabefeldes auch bei editlog. Außerdem entfällt ein Template komplett, d.h. man hat später beim Redesign weniger Arbeit.

Den Test bekomme ich notfalls auch alleine hin.

following5 commented:
Wollte das eigentlich heute fertig testen und noch ein paar Details in der Codestruktur geradeziehen, aber warte jetzt erst mal ab wie es hier weitergeht.

following5 commented:
Da die Entwicklungsarbeit hier schon zu 90% getan ist, würde durch Wegwerfen und Neuimplementation auf Symfony-Basis auf jeden Fall mehr Aufwand entstehen.

following5 committed:
refactored editlog.php and fixed some old and new issues

following5 commented:
Bevor ich mich ein zweites mal ransetzen und neu reindenken muss, mache ich das hier nun doch schon fertig.

Beim Refactoring von editlog.php sind noch einige Problemchen und veraltetes Zeug aufgefallen, das ist nun alles geradegezogen. Insgesamt nun 270 eingesparte Code- und Templatezeilen und ca. 10 funktionelle Verbesserungen in editlog.

following5 committed:
adjustments after editlog overhaul

- don't ask for listing-ok confirmation when editing a log and
    - there is a newer listing-state log, or
    - the cache's current LO state was generated by the edited log, or
    - log already had listing-ok flag (i.e. it was previously confirmed)

- disable status log types when editing old non-status log
- never disable NM and LO selection lists for admin logs
- fixed log password handling
- consildate some common log/editlog code
- log_cache.tpl formatting
- gray rating checkbox label when checkbox is disabled

following5 commented:
* Wartungsstatus-Markierungen beim Editieren von Logs änderbar (Redmine #1050)
* Usability-Verbesserungen aus log nach editlog übernommen
* Funktionalität, Bedienung und Design von log und editlog sind nun konsistent
* ein paar kleinere Problemchen in editlog behobem, Fehlerhanding verbessert
* gammligen alten editlog-Code aufgeräumt
* etwas Refactoring im Log-Template
* redundanten Template-Code entsorgt

following5 commented:
Von mir aus ist das Ganze nun hinreichend getestet, um es freizugeben. Dass wesentliche neue Bugs drin sind, halte ich für unwahrscheinlich.

Die eine oder andere Unstimmigkeit im Detail kann ich nicht ganz ausschließen, allerdings habe ich davon auch einige behoben.

In der Funktionalität, Usability und Codequalität ist es ein deutlicher Fortschritt. Es ist auch ein großer Schritt in Richtung Zusammenführung von log.php und editlog.php; das würde nochmal ca. 200 redundante Codezeilen einsparen.

following5 commented:
Die commit-Aufteilung würde ich gerne so lassen, dann ist es besser nachzuvollziehen und zu debuggen. Die Zwischenschritte sind alle voll lauffähig.

following5 commented:
Durch diese Änderung wird auch die Umsetzung von Redmine #1023 einfacher.

following5 commented:
rebased

Falls ich noch irgendwas tun kann für diesen PR, lass es mich bitte wissen. Bugs sind mir auch beim Neutest nach Rebase keine mehr aufgefallen. Wenn's sein muss setze ich mich auch nochmal nen halben Tag hin und reviewe und teste alles nochmal. Denke das ist's wert.

Ist auf jeden Fall viel stabiler als damals die Portierung von log.php auf lib2, und damals ist auch nix schlimmes passiert. ;)

teiling88 commented on htdocs/editlog.php:
```diff
@@ -15,527 +15,432 @@
-        0,
-        $cacheId
+// fuer alte Versionen von OCProp
+if (isset($_POST['submit']) && !isset($_POST['version2'])) {
```
Wenn das If Statement nur für OcProp da ist - bitte entfernen.

following5 replied:
Das if-Statement verhindert, dass alte Ocprop-Versionen fehlerhafte Logs hochladen. Sicher, dass das raus soll?

teiling88 replied:
Ja - ocprop wird blockiert da nur eine neue Version den Server nicht mit Anfragen überfluten würde. Wir mussten daher alle alten Versionen sperren. Den Patch hab ich schon glaube vor einem drei viertel Jahr eingereicht. Leider ist nichts passiert. 

Daher baue ich aktuell ein Möglichkeit in der man die founds.gpx von geocaching.com hochladen und damit seine Funde abgleichen kann. Das gleiche baue ich auch für die Caches.

following5 replied:
> ocprop wird blockiert da nur eine neue Version den Server nicht mit Anfragen überfluten würde.

Fällt mir eben noch ein: Anfragefluten gab es bei Ocprop immer. Also wenn zwei GC-Logger mit tausenden von Funden zufällig gleichzeitig ihren (z.B.) quartalsweisen Abgleich machen, dann ist der Server die nächsten 1-2 Stunden hauptsächlich mit Ocprop beschäftigt. Konnte man aber mit leben, die Seite blieb bedienbar.

Nur einmal wure es echt kritisch, weil jemand Ocprop täglich hat laufen lassen; den haben wir dann angemailt und danach war gut.

teiling88 commented on htdocs/editlog.php:
```diff
@@ -15,527 +15,432 @@
+}
+
+// fuer alte Versionen von OCProp
+if ($descMode != 1 && isset($_POST['submit']) && !isset($_POST['version2'])) {
```
Wenn das If Statement nur für OcProp da ist - bitte entfernen.

following5 replied:
siehe oben

teiling88 commented on htdocs/lib2/logic/cache.class.php:
```diff
@@ -272,8 +272,7 @@ public function getListingOutdatedLogUrl()
              FROM `cache_logs`
-             WHERE `cache_id`='&1'
-             AND `listing_outdated`>0
+             WHERE `cache_id`='&1' AND `listing_outdated`>0
```
Bitte nicht wieder in eine Zeile rücken.

teiling88 commented on htdocs/lib2/logic/cache.class.php:
```diff
@@ -1004,4 +1031,23 @@ public function statusUserLogAllowed()
+     * @param $logId
+     * @return bool
+     */
+    function isLatestLog($logId)
```
**public** function

teiling88 commented on htdocs/lib2/logic/cache.class.php:
```diff
@@ -1004,4 +1031,23 @@ public function statusUserLogAllowed()
+    /**
+     * @param $cacheId
+     * @param $logId
+     * @return bool
```
phpdoc ist nicht aktuell

teiling88 commented on htdocs/log.php:
```diff
@@ -206,34 +202,11 @@
+    $validate['dateOk'] = cachelog::validateDate(
+        $logDateYear, $logDateMonth, $logDateDay,
+        $logTimeHour, $logTimeMinute,
+        isset($_POST['submitform'])
```
Wenn Du das PSR-2 konform machen möchtest muss jeder Parameter in eine Zeile.

teiling88 commented:
Ein paar Kleinigkeiten - dann kann es gemerged werden.

following5 committed:
edilog code style, comments and improved listing-outdated cofirmation

following5 committed:
removed log/editlog code related to old Ocprop versions

following5 commented:
Ist soweit erledigt. Getrennt Commits, weil das eine eine Nachbesserung zu meinem Change ist, das andere eine davon unabhängige Änderung.
